### PR TITLE
added spinner and validation to sign in page 

### PR DESCRIPTION
**Pull request by pdybowski**
image
image

### CODE REVIEW

File: frontend/src/pages/Common/LogIn/LogIn.tsx [hunk 1]
```diff
@@ -22,43 +23,70 @@ export interface LogInProps {
...
-  const [password, setPassword] = useState('')
+  const [form, setForm] = useState({ email: '', password: '' })
+  const [invalidEmail, setInvalidEmail] = useState(false)
+  const [invalidPasswd, setInvalidPasswd] = useState(false)
+  const [loading, setLoading] = useState(false)
 
   const history = useHistory()
   const routeChange = () => {
-    let path = `/home`
+    const path = `/home`
```
Comment (hoolek77) [inline frontend/src/pages/Common/LogIn/LogIn.tsx:37]: No need for backticks there

File: frontend/src/pages/Common/LogIn/LogIn.tsx [hunk 2]
```diff
@@ -68,38 +96,45 @@ export default function SignIn(props: LogInProps) {
...
+                variant="contained"
+                color="primary"
+                className={classes.submit}
+                data-testid="li-button"
+              >
+                Sign In
+              </Button>
+            ) : (
+              <CircularProgress color="primary" className={classes.submit} />
+            )}
```
Comment (hoolek77) [inline frontend/src/pages/Common/LogIn/LogIn.tsx:137]: In my opinion, you can move the loader inside the button and replace text with loader, while it's loading. If so, add prop 'disabled={loading}' to a button.

**Review by hoolek77 — CHANGES_REQUESTED** [review]

File: frontend/src/pages/Common/LogIn/LogIn.tsx (hunk 2 above)
Comment (pdybowski) [reply frontend/src/pages/Common/LogIn/LogIn.tsx:137]: this is the same convention as that in the register page. It actually depends on the business, project manager or The Boss #Konrad

File: frontend/src/pages/Common/LogIn/LogIn.tsx (hunk 1 above)
Comment (pdybowski) [reply frontend/src/pages/Common/LogIn/LogIn.tsx:37]: not my code, but i can change it :p

File: frontend/src/pages/Common/LogIn/LogIn.tsx (hunk 2 above)
Comment (pdybowski) [reply frontend/src/pages/Common/LogIn/LogIn.tsx:137]: I should spend more time to find proper loading gif that could work inside disabled button so not sure if it's necessary for now

**Review by hoolek77 — APPROVED** [review]

**Pull request merged**